capture/configs/aes_sca_cw310.yaml

```diff
@@ -4,11 +4,11 @@ target:
   force_program_bitstream: True
   fw_bin: "../objs/aes_serial_fpga_cw310.bin"
   pll_frequency: 100000000
+  target_freq: 24000000
```

Comment:
In my view, we should have the target_clk_mult value instead of the pll_frequency value here. And we should say in a comment directly above target_clk_mult that target_clk_mult is not to be changed by the user but that this is a value hard-coded in the bitstream. The user should then configure the target_freq value, and based on this value and the  target_clk_mult  we internally determine the pll_frequency and afterwards the sampling_rate (where sampling_rate = pll_frequency * int).

I think there is little value in being able to specify both pll_frequency and target_freq because there is a fixed relationship between the two.

Does this sound reasonable or is there a better way to handle this maybe?

Reply:
As all three values are related, e.g., target_freq

Reply:
The only value fixed by the bitstream is `target_clk_mult`. `pll_frequency` is what we configure via the CW API and `target_freq` is the resulting target frequency determined by your equation.

What I am trying to avoid is that people for whatever reason configure e.g. 50 MHZ `pll_frequency` and 10 MHz `target_freq`, and then spend time debugging why it's not working as expected. It should be crystal clear for users which value to tune and which values to leave alone.

As for the sampling rate: It cannot arbitrarily be chosen on Husky. There is always the fixed relationship `sampling_rate` = `pll_frequency * adc.mul`  where `adc.mul` must be an integer. So everything is kind of linked together. Luckily, we have more freedom with other scopes. Again, what I am trying to avoid is that people configure values that don't work and then wonder why. We will spend time supporting them, debugging their issues, and explain to them how things work.

Reply:
Make sense, `target_clk_mult` and `target_freq `are now configurable.

Reply:
@vogelpi now it should be implemented as discussed. `sampling_rate `can be optional, then the scope driver automatically calculates the maximum available sampling rate and takes care that adc.mul is a int.

Reply:
This is perfect, thanks @nasahlpa !